api/end_device.proto

```diff
@@ -116,6 +114,11 @@ message MACParameters {
   // Whether downlink dwell time is set (400ms).
   // If this field is not set, then the value is either unknown or irrelevant(Network Server cannot modify it).
   google.protobuf.BoolValue downlink_dwell_time = 21;
+
+  // ADR: number of messages to wait before setting ADRAckReq.
+  ADRAckLimitExponent adr_ack_limit = 22 [(gogoproto.customname) = "ADRAckLimit", (validate.rules).enum.defined_only = true];
```

Comment:
This breaks API compatibility for our JSON API (if they could read this field). It was a number, now it's a string.

Reply:
It was not possible to set these via API and I'm quite sure console does not use these. @bafonins @kschiffer can you confirm?

Reply:
This isn't about the console using the fields or not, this is about breaking API for which we promised stability. If clients were able to GET these fields before, and now suddenly receive a string instead of a number, we might break those clients if they rely on a field being a specific type. This is the case for any hypothetical Go client that uses JSON encoding instead of Protobuf.

Reply:
With our compatibility promise in mind, I think the right way to do this would be to add the new fields with a different name, like `adr_ack_limit_exponent`, while keeping the old ones and fill the value of `adr_ack_limit` with the corresponding number from the `_exponent`.

Since clients were only allowed to GET these fields, and not to SET them, we only have to implement the SET logic for the `_exponent` fields.